microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -0,0 +1,252 @@
+        self._execute_expect_zero(node, "mkdir -p /mnt/huge")
+        self._execute_expect_zero(node, "mkdir -p /mnt/huge-1G")
+        self._execute_expect_zero(node, "mount -t hugetlbfs nodev /mnt/huge")
+        self._execute_expect_zero(
+            node, "mount -t hugetlbfs nodev /mnt/huge-1G -o 'pagesize=1G'"
+        )
+
+    def _hugepages_enable(self, node: Node) -> None:
+        self._execute_expect_zero(
+            node,
+            "echo 4096 > /sys/devices/system/node/node0/hugepages/hugepages-2048kB/nr_hugepages",
+        )
+        self._execute_expect_zero(
+            node,
+            "echo 1 > /sys/devices/system/node/node0/hugepages/hugepages-1048576kB/nr_hugepages",
+        )
+        result = node.execute("grep -i huge /proc/meminfo && ls /mnt/", shell=True)
+        self.log.info(f"hugepages status \n{result.stdout}")
+
+    def _install_dpdk(self, node: Node) -> None:
+        git_tool = node.tools[Git]
+        git_tool.clone(self._dpdk_github, cwd=node.working_path)
+        dpdk_path = node.working_path.joinpath("dpdk")
+        self._execute_expect_zero_with_path(node, "meson build", dpdk_path)
+        self.log.info(node.execute("ls -la", cwd=dpdk_path).stdout)
+        dpdk_build_path = dpdk_path.joinpath("build")
+        self._execute_expect_zero_with_path(node, "which ninja", dpdk_build_path)
+        self._execute_expect_zero_with_path(node, "ninja", dpdk_build_path)
+        self._execute_expect_zero_with_path(node, "ninja install", dpdk_build_path)
+        self._execute_expect_zero_with_path(node, "ldconfig", dpdk_build_path)
+
+    def _get_nic_names(self, node: Node) -> List[str]:
+        result = node.execute(
+            " ls /sys/class/net/ | grep -Ev $(ls /sys/devices/virtual/net)",
+            shell=True,
+            sudo=True,
+        )
+        nic_names = result.stdout.split("\r\n")
+        for item in nic_names:
+            assert_that(item).is_not_equal_to("").described_as(
+                "nic name could not be found"
+            )
+        self.log.info(f"network devices: {nic_names}")
+        return nic_names
+
+    def _get_nic_device(self, node: Node, nic_name: str) -> str:
+        nic_device = self._execute_expect_zero(
+            node, f"readlink /sys/class/net/{nic_name}/device"
+        )
+        base_device_name = self._execute_expect_zero(node, f"basename {nic_device}")
+        assert_that(base_device_name).is_not_equal_to("")
+        return base_device_name
+
+    def _get_primary_secondary_pairings(
+        self, node: Node, nic_list: List[str]
+    ) -> Dict[str, str]:
+        # Identify which nics are slaved to master devices.
+        # This should be really simple with /usr/bin/ip but experience shows
+        # some platforms are buggy and require another method
+        master_nics = dict()
+        result = node.execute(
+            "echo 'export PATH=$PATH:/usr/sbin' >> ~/.bashrc", shell=True
+        )
+        self.log.info(result.stdout)
```

Comment:
The information is in debug level already. And don't output too much information on info level. The "info" level is a story for looking the flow. The debug information should be in "debug" level.